src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -1428,6 +1428,9 @@ private void handleSwitch(JCTree swtch, JCExpression selector, List<JCCase> case
                     code.put4(caseidx + 4, offsets[i]);
                 }
             }
+
+            // Emit line position for the end of the switch
+            code.statBegin(TreeInfo.endPos(swtch));
```

Comment:
shouldn't this change affect already existing tests like: RuleSwitchBreaks.java in the same folder: test/langtools/tools/javac/classfiles/attributes/LineNumberTable/?

Reply:
no I was just wondering why the issue didn't appear in the previous test, but what you are saying could be the explanation

Reply:
What do you think about this? https://github.com/openjdk/jdk/pull/17772/commits/a836d976e56c624882ae793ee89b1146c9fd4352

(also notice the newly added test case. For the `switch` statement the line 8 is not included and for the `switch` expression it is.)

Reply:
yep looks good, thanks